hypervisor/arch/x86/configs/apl-mrb/ve820.c

```diff
@@ -7,33 +7,33 @@
 #include <e820.h>
 
 const struct e820_entry ve820_entry[E820_MAX_ENTRIES] = {
-	{	/* 0 to mptable */
+	{	/* usable RAM under 1MB */
 		.baseaddr = 0x0UL,
-		.length   = 0xEFFFFUL,
+		.length   = 0xF0000UL,		/* 960KB */
 		.type     = E820_TYPE_RAM
 	},
 
-	{	/* mptable 65536U */
-		.baseaddr = 0xF0000UL,
-		.length   = 0x10000UL,
+	{	/* mptable */
+		.baseaddr = 0xF0000UL,		/* 960KB */
+		.length   = 0x10000UL,		/* 16KB */
 		.type     = E820_TYPE_RESERVED
 	},
 
-	{	/* mptable to lowmem */
-		.baseaddr = 0x100000UL,
-		.length   = 0x1FF00000UL,
+	{	/* lowmem */
+		.baseaddr = 0x200000UL,		/* 2MB */
+		.length   = 0x1FF00000UL,	/* 511MB */
 		.type     = E820_TYPE_RAM
 	},
 
-	{	/* lowmem to PCI hole */
-		.baseaddr = 0x20000000UL,
-		.length   = 0xa0000000UL,
+	{	/* between lowmem and PCI hole */
+		.baseaddr = 0x20100000UL,	/* 513MB */
+		.length   = 0x9FF00000UL,	/* 2559GB */
```

Comment:
should be 256GB, not?

Reply:
Because of the 1MB memory hole, the lowmem entry is 2MB-- 0x20100000 (512MB + 1MB). If we keep this entry as 2560MB, it will eat 1MB into the PCI hole (0xc0000000). So I cut 1MB from this reserved entry and keep PCI hole clean.